### PR TITLE
3 more tasks finished

### PR DESCRIPTION
**Pull request by ty-shaikh**
1. User can edit a comment they left
2. Posts on the feed page have truncated bodies
3. User dashboard pulls in comments and share data

### CODE REVIEW

File: app/views/shares/index.html.erb [hunk 1]
```diff
@@ -3,13 +3,25 @@
...
-    <div class="handle"><%= current_user.name %></div>
-    <%= link_to share.title, share_path(share) %>
-    <p>
-      <%= linkify_hashtags(markdown(share.body)) %>
-    </p>
-    <%= pluralize(share.votes.count, "votes") %>
-    
+    <div class="handle">
+      <h4><%= link_to share.title, share_path(share) %></h4>
+      <p><%= linkify_hashtags(truncate(share.body, length: 100, separator: ' ')) %></p>
```
Comment (kenmazaika) [inline app/views/shares/index.html.erb]: This looks like it removes the markdown conversion that we had before.  I think we ultimately want both, although it's fairly complicated.

I think this gem may do it:
https://github.com/ianwhite/truncate_html

File: app/views/shares/show.html.erb [hunk 2]
```diff
@@ -33,6 +40,14 @@
       <blockquote>
         <%= comment.body %>
         <small><%= comment.user.email %></small>
+        <% if comment_owner?(comment) %>
```
Comment (kenmazaika) [inline app/views/shares/show.html.erb]: I like the use of the helper here.

**Comment by kenmazaika** [comment]
Had 1 comment I think we should look into about using markdown.

**Comment by kenmazaika** [comment]
Nice!

**Comment by ty-shaikh** [comment]
Ok, I will update the code. My local repo was not up-to-date with the upstream, so did not see the markdown helper.

**Comment by kenmazaika** [comment]
It looks like some of the tests are failing.  It looks like I may have pushed the merge button a little too quickly.

```

1) CommentsController EDIT comment should only allow the comment creator to edit the body
Failure/Error: expect(response).to have_http_status(:forbidden)
  expected the response to have status code :forbidden (403) but it was :found (302)
# ./spec/controllers/comments_controller_spec.rb:41:in `block (3 levels) in <top (required)>'

2) CommentsController UPDATE comment should only allow the comment creator to edit the body
Failure/Error: expect(response).to have_http_status(:forbidden)
  expected the response to have status code :forbidden (403) but it was :found (302)
# ./spec/controllers/comments_controller_spec.rb:64:in `block (3 levels) in <top (required)>'

3) SharesController shares#show action should return a 404 error if the share is not found
Failure/Error: @share = Share.find(params[:id])

ActiveRecord::RecordNotFound:
  Couldn't find Share with 'id'=NOTHING HERE
# ./app/controllers/shares_controller.rb:67:in `set_share'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `block in process'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `catch'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `_catch_warden'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `process'
# ./spec/controllers/shares_controller_spec.rb:59:in `block (3 levels) in <top (required)>'

4) SharesController shares#edit should return a 404 error message if the share is not found
Failure/Error: @share = Share.find(params[:id])

ActiveRecord::RecordNotFound:
  Couldn't find Share with 'id'=NOT FOUND
# ./app/controllers/shares_controller.rb:67:in `set_share'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `block in process'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `catch'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `_catch_warden'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `process'
# ./spec/controllers/shares_controller_spec.rb:90:in `block (3 levels) in <top (required)>'

5) SharesController shares#update should have http 404 error if the share cannot be found
Failure/Error: @share = Share.find(params[:id])

ActiveRecord::RecordNotFound:
  Couldn't find Share with 'id'=DOES NOT EXIST
# ./app/controllers/shares_controller.rb:67:in `set_share'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `block in process'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `catch'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `_catch_warden'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `process'
# ./spec/controllers/shares_controller_spec.rb:120:in `block (3 levels) in <top (required)>'

6) SharesController shares#destroy should return a 404 message if we cannot find a share with the id specified
Failure/Error: @share = Share.find(params[:id])

ActiveRecord::RecordNotFound:
  Couldn't find Share with 'id'=SHARE IS NOT HERE
# ./app/controllers/shares_controller.rb:67:in `set_share'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `block in process'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `catch'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:72:in `_catch_warden'
# /home/rof/cache/bundler/ruby/2.2.0/gems/devise-3.5.10/lib/devise/test_helpers.rb:19:in `process'
# ./spec/controllers/shares_controller_spec.rb:157:in `block (3 levels) in <top (required)>'

Finished in 1.29 seconds (files took 3.61 seconds to load)
38 examples, 6 failures

Failed examples:

rspec ./spec/controllers/comments_controller_spec.rb:36 # CommentsController EDIT comment should only allow the comment creator to edit the body
rspec ./spec/controllers/comments_controller_spec.rb:59 # CommentsController UPDATE comment should only allow the comment creator to edit the body
rspec ./spec/controllers/shares_controller_spec.rb:58 # SharesController shares#show action should return a 404 error if the share is not found
rspec ./spec/controllers/shares_controller_spec.rb:87 # SharesController shares#edit should return a 404 error message if the share is not found
rspec ./spec/controllers/shares_controller_spec.rb:117 # SharesController shares#update should have http 404 error if the share cannot be found
rspec ./spec/controllers/shares_controller_spec.rb:154 # SharesController shares#destroy should return a 404 message if we cannot find a share with the id specified
```

**Comment by kenmazaika** [comment]
Should we revert merging this PR or should we fix the tests?

**Comment by ty-shaikh** [comment]
I'll fix it. The controller had my old code: Share.find(params[:id]) instead of Share.find_by_id(params[:id])

**Pull request merged**